packages/edit-post/src/components/header/fullscreen-mode-close/index.js

```diff
@@ -11,33 +11,43 @@ import { IconButton, Toolbar } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
 
-function FullscreenModeClose( { isActive, postType } ) {
-	if ( ! isActive || ! postType ) {
+function FullscreenModeClose( { isActive, postType, isAlwaysShown, isLabeled } ) {
+	if ( ! ( isActive || isAlwaysShown ) || ! postType ) {
```

Comment:
You could also use the  `ifCondition` HOC to hide the post based on these conditions. See something similar here: https://github.com/WordPress/gutenberg/pull/17117/files#diff-6bc1caaad4f05a3668923c347ab3834eR57

You basically do ifCondition( props => something); and, something should be true if you want the component to display, and false if you want the component to be hidden.